### PR TITLE
Tdr 487 initial setup

### PR DESCRIPTION
**Pull request by TomJKing**

### CODE REVIEW

**Comment by MancunianSam** [comment]
Looks good so far. 
I would say that you can put this into the same VPC as jenkins. Maybe to avoid confusion we could rename the tdr-jenkins-vpc-mgmt to tdr-vpc-mgmt or something similar although we don't have to.
This would mean that you wouldn't need `flowlogs.tf` and `vpc.tf` in the modules directory which should tidy things up a bit.

The other thing is that there's a lot of terraform code in the `modules/grafana` directory. Would it be better to move as much as possible into the `tdr-terraform-modules` repo?
`ecr.tf` can be moved into `tdr-terraform-modules/ecs` in its own file with its own count and `route53.tf` can be replaced with calling the module from `root.tf`
I'm not sure about the other files but if we shuffle as much as possible onto `tdr-terraform-modules` then it should be a bit tidier.

Last small thing is that we're not (or I'm not anyway) using the `data "template_file"` blocks but are using `templatefile("${path.module}/templates/checksum_lambda.json.tpl", { environment = local.environment ...` instead. This is personal preference though so feel free to ignore it.

File: terraform/modules/grafana/security.tf [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  tags = merge(
+    var.common_tags,
+    map("Name", "tdr-grafana-ecs-task-security-group-${var.environment}")
+  )
+}
+
+resource "aws_security_group" "database" {
+  name        = "tdr-grafana-database-security-group-${var.environment}"
+  description = "Allow inbound access from the Grafana load balancer only"
+  vpc_id      = var.vpc_id
```
Comment (MancunianSam) [inline terraform/modules/grafana/security.tf:56]: Should this say "access to and from the ECS task only" or something like that ?

File: terraform/modules/grafana/subnets.tf [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  count = var.az_count
+  tags = {
+    Name = "nat-gateway-${count.index}-mgmt-mgmt"
+  }
+}
+
+# Bring default security group under Terraform management and remove all rules
+resource "aws_default_security_group" "default" {
+  vpc_id = var.vpc_id
+}
```
Comment (MancunianSam) [inline terraform/modules/grafana/subnets.tf]: Do you need this? There's a resource for the default security group already being managed in the Jenkins terraform and there's only one default security group per VPC.

File: terraform/root_variables.tf [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+variable "domain_name" {
+  description = "fully qualified domain name for the service"
+  default     = "grafana.tdr-management.nationalarchives.gov.uk"
+}
+
+variable "function" {
+  description = "forms the second part of the bucket name, eg. upload"
+  default     = "grafana"
+}
+
```
Comment (MancunianSam) [inline terraform/root_variables.tf:23]: For the `function` and `project` variables, they both mention bucket names but they're in the root_variables here so it's not clear what they refer to. Maybe the function one could reference the alb. I don't think you need to mention a bucket at all for the `project` variable, just leave it at "abbreviation for the project" What do you think?

Comment (TomJKing) [reply terraform/root_variables.tf:23]: Reworded

File: terraform/modules/grafana/security.tf (hunk 1 above)
Comment (TomJKing) [reply terraform/modules/grafana/security.tf:56]: Reworded

File: terraform/modules/grafana/subnets.tf (hunk 2 above)
Comment (TomJKing) [reply terraform/modules/grafana/subnets.tf]: removed

**Review by MancunianSam — APPROVED** [review]

**Pull request merged**